python_scripts/select_architectures.py

```diff
@@ -148,23 +153,22 @@ def _main():
             if key.startswith("deps-"):
                 print(f"non-hermetic build, disabling {key} task")
                 architecture_decision[key] = "localhost"
-    if arches == ['noarch']:
-        # when exclusivearch
-        if arches['exclusivearch']:
-            build_arches = arches['exclusivearch']
-            # remove excludeArches
-            build_arches = list(set(build_arches) - set(arches['excludearch']))
-        else:
-            # default build arches
-            build_arches = ['x86_64', 'i686', 'aarch64', 's390x', 'ppc64le']
-            # build arches without excludeArch
-            build_arches = list(set(build_arches) - set(arches['excludearch']))
-        selected_architectures = [random.choice(build_arches)]
+
+
+    build_arches = allowed_architectures
+    if arches['exclusivearch']:
+        build_arches = list(set(build_arches) & set(arches["exclusivearch"]))
+    if arches['excludearch']:
+        build_arches = list(build_arches - set(arches["excludearch"]))
+    if arches['buildarch'] == ['noarch']:
+        allowed_architectures = [random.choice(build_arches)]
+    else:
+        allowed_architectures = build_arches
```

Comment:
I'd slightly prefer keeping `allowed_architectures` as _input only_ variable, and make `selected_architectures` the _output only_.  And I'd prefer working with set() like `a &= b` (setting them to be of type set() as soon as possible, and then avoid further retype).